packages/react-app/src/components/actions/increaseStake.js

```diff
@@ -0,0 +1,123 @@
+import React, { useContext, useState, useEffect } from 'react'
+import { Container, Row, Col, Form } from 'react-bootstrap/';
+import { TypeOver, PendingButton, Slider, Grey, Blue, NuStakeAllocator, CircleQ, DataRow, Period } from '@project/react-app/src/components'
+
+import { formatWei } from '@project/react-app/src/constants'
+
+import { Context, ContractCaller, setNUAllowance } from '@project/react-app/src/services'
+
+
+export const IncreaseStake = (props) => {
+
+    const context = useContext(Context)
+    const { web3, contracts, account } = context.wallet
+
+    const substake = props.substake
+
+    const [nuAllocated, setNuAllocation] = useState()
+    const [maxNULimit] = useState(web3.utils.toBN(context.availableNU.get || 0))
+    const [AllocationValid, setAllocationValid] = useState(true)
+    const [invalidMessage, setInvalidMessage] = useState()
+    const [approvingNUspend, setApprovingNUspend] = useState(false)
+    const [iAmDisclaimed, setIAmDisclaimed] = useState(false)
+
+    const handleAction = (e) => {
+
+        const hex = web3.utils.numberToHex(0)
+        ContractCaller(
```

Comment:
@vepkenez there are two modes of operation for increase
- increase stake using unlocked tokens already in StakingEscrow (`lock_and_increase`)
- increase stake using tokens in wallet (`deposit_and_increase`)

Do we handle both cases? See https://github.com/nucypher/nucypher/blob/main/nucypher/blockchain/eth/actors.py#L705 for context.

Reply:
Nope.  I was not aware of `lock_and_increase`.
I guess currently we only increase stakes from wallet NU.  

Personally I'm fine with this.  It seems complicated to have to manage all these pools locked/unlocked/inside/outside the contract.

Reply:
It's not too bad (logically, not sure about the react-js needs 😅 ):
1. User chooses to increase using tokens from wallet or unlocked tokens.
2. If user chooses unlocked tokens, instead of obtaining the available balance from the account, you obtain the balance from StakingEscrow (`calculate_staking_reward`) - see https://github.com/nucypher/nucypher/blob/main/nucypher/blockchain/eth/actors.py#L693
3. Call `lockAndIncrease` - the parameters are the same as `depositAndIncrease`.

This is the use case when people want to increase their stake using staking rewards not previously withdrawn. In any case, something to consider - perhaps we just create a separate issue for tracking, and if there is sufficient demand for it, we can implement it. wdyt?

Reply:
Sure, yeah that doesn't sound too bad.  Can a _new_ stake be initiated from unlocked rewards?  It would definitely make sense to me to allow users to select a `"source of funds": unlocked rewards | wallet` for that as well.

Reply:
Yep, same philosophy just `lockAndCreate` instead of `deposit` - see https://github.com/nucypher/nucypher/blob/main/nucypher/blockchain/eth/actors.py#L625.